### PR TITLE
Wrap firebase operations in transactions

### PR DESCRIPTION
**Pull request by lumisphere902**
I don't really have a method of testing how this affects Firebase, but it at least compiles and runs
- Wrapped DB operations in transactions
- Add a bit more error handling

### CODE REVIEW

**Comment by malsf21** [comment]
As a heads up @lumisphere902, if you update the head branch for a PR (i.e. by pushing a new commit), the PR automatically updates - no need to close a PR and open a new one! If you want to erase history, you can also force-push or rebase instead!

**Comment by lumisphere902** [comment]
> As a heads up @lumisphere902, if you update the head branch for a PR (i.e. by pushing a new commit), the PR automatically updates - no need to close a PR and open a new one! If you want to erase history, you can also force-push or rebase instead!
> 
> It's good practice to leave past history up so we can discuss is/use it for code review. NBD though 😄

It closed itself for some reason; not sure why lol

File: db/db.go [hunk 1]
```diff
@@ -280,26 +302,34 @@ func (d *DB) MakeAlias(ctx context.Context, uid string, path string) (string, er
...
-	for snap.Exists() == true {
-
-		aid++
-		if aid >= 0xFFFFFFFFF {
-			aid = 0
+	err := d.RunTransaction(ctx, func(ctx context.Context, tx *firestore.Transaction) error {
+		doc, err := tx.Get(col.Doc(wid))
+		if err != nil {
+			return err
 		}
```
Comment (krashanoff) [inline db/db.go:310]: The firestore library spec states that err will be set if the document doesn't exist. In this case, we are expecting the document not to exist, so one would expect err to contain a message complaining about the document with their supplied rpc code `NotFound`. Indeed, this is the case, so the function was returning early with an error that we expected, which fails the tests. This same error occurs L324..326

My quick suggestion as to how to fix this is to be more specific in the error checks here. Something to the tune of `doc == nil || (err != nil && doc.Exists())` might do the trick, checking if we have an error despite the document existing.

Aside from this, LGTM!

**Comment by krashanoff** [comment]
Hey Timothy, just a reminder that the bug you fixed in your most recent commit is also on L324. Also, using my provided condition to the letter will not work under the condition that doc is nil and err is also nil. Though this situation shouldn't arise with Go's firestore lib, it does produce the edge case that we faultily return an empty error without creating the alias -- just something to consider. Thanks again and my apologies for being a stickler.

**Comment by lumisphere902** [comment]
> Hey Timothy, just a reminder that the bug you fixed in your most recent commit is also on L324. Also, using my provided condition to the letter will not work under the condition that doc is nil and err is also nil. Though this situation shouldn't arise with Go's firestore lib, it does produce the edge case that we faultily return an empty error without creating the alias -- just something to consider. Thanks again and my apologies for being a stickler.

I added it in the other place, but I'm not sure what the expected behavior is for not finding a document and also not throwing an error.

**Pull request merged**